client/src/hooks/members/useMemberDetails.ts

```diff
@@ -0,0 +1,130 @@
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { AxiosError } from "axios";
+import { MemberDetailsDto, MemberImage } from "../../types/membersTypes";
+import { authReducerActionType } from "../../types/globalTypes";
+import {
+  getMemberDetails,
+  getMemberImage,
+  updateMemberDetails as updateMemberDetailsApi,
+  saveMemberImage as saveMemberImageApi,
+} from "../../api/members";
+import { showErrorMessage, showSuccessMessage } from "../../utils/toastUtils";
+import { useContext } from "react";
+import { AuthContext } from "../../context/auth-context/AuthContext";
+
+/**
+ * Hook that handles the members api calls, uses react-query
+ * @returns The member details, a function to update the details of a member, the member image and a function to save the image of a member
+ */
+const useMemberDetails = (memberID: number) => {
+  const { dispatchAuth } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+
+  // ############
+  // GET QUERIES
+  // ############
+
+  // getMemberDetails query
+  const {
+    data: memberDetailsData,
+    isLoading: isMemberDetailsLoading,
+    isError: isMemberDetailsError,
+  } = useQuery({
+    queryKey: ["memberDetails", memberID],
+    queryFn: () => getMemberDetails(memberID),
+  });
+
+  const memberDetails = memberDetailsData && (memberDetailsData.data as MemberDetailsDto);
```

Comment:
This is not very intuitive.
Suggestion:

```suggestion
  const memberDetails = memberDetailsData?.data || null;
```